designs/ClientDevProcess.md

```diff
@@ -0,0 +1,145 @@
+- Author: [Kat Marchan](https://github.com/zkat)
+
+# Issue
+
+[#8994](https://github.com/NuGet/Home/issues/8994)
+
+# Background
+
+For a while now, the NuGet Client team has been using [ZenHub](https://zenhub.com) as part of its development process. At the same time, while ZenHub provides a lot of features, the team isn't taking advantage of some of the nicer ones, such as its charting capabilities and its Roadmap feature. It also has other features, like Blockers and Epics, which the team does use but could be using more often, or differently.
+
+This proposal is aimed at overhauling the Client Team's usage of ZenHub and further streamline its public development process to increase velocity, as well as increase visibility into what's being done, and what needs to be done. It also describes existing processes, so this document may be used as a reference for development flow.
+
+# Proposal
+
+## OKR Tracking
+
+- Link: https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/roadmap?repos=29996513
+
+One of the main additions in this proposal is that it adds explicit tracking of OKRs into our ZenHub process. The way this works is by using ZenHub's Roadmap feature.
+
+Each quarter, a new Project should be created for each quarterly Objective, and labeled as such. Inside each Project should be any Epics created for those particular KRs, with dates adjusted appropriately. Additionally, a "Non-KR Work" Project should be created, with any non-KR epics, such as bugfixes and such. Ideally, all work will exist under some quarterly epic and be displayed on the roadmap.
+
+## Epic Tracking
+
+The Client Team uses Epics to track high-level status of various issues. Epics should be created for tracking large verticals of work, or when there's many disparate subtasks that will be involved and potentially assigned to multiple people, or involve multiple PRs to complete. As described in [OKR Tracking](#okr-tracking), they should also be created to track specific KRs.
+
+Another way to think about Epics is: if there is a customer-facing feature, an Epic should be the parent of all tasks from spec'ing to writing the actual code to test to insertion to shipping and writing docs.
+
+Concretely, these are GitHub issues tagged as Epic, and they navigate the board slightly different from regular issues:
+
+1. When any issue under an Epic is In Progress, the associated Epic should be moved to the In Progress pipeline as well.
+2. Epics should remain under "In Review" or "Validating" until the related change is available to customers. That is, they shouldn't be closed until released.
+
+Tracking Epics in this way allows customers and stakeholders to see a more fine-grained view of the status of any KR work by going to the Board and [filtering by Epics and hiding subtasks](https://github.com/NuGet/Home#workspaces/nuget-client-team-55aec9a240305cf007585881/board?epics:settings=epicsOnly&filterLogic=any&repos=29996513).

```

Comment:
We should consider defining an issue template for Epics (a future iteration of this, let's not change too much). 

Similarly, I've been thinking we should have an issue template for binary code flow etc.

Reply:
We should probably have a proposal for overhauling our issue templating in general. GitHub has improved the way this works a lot in recent times, and you can get buttons and different types of templates now!